R/foreign-keys.R

```diff
@@ -164,27 +163,30 @@ cdm_rm_fk <- function(dm, table, column, ref_table) {
 #' Find foreign key candidates in a table
 #'
 #' Determine which columns would be good candidates to be used as foreign keys of a table,
-#' and reference the primary key column of another [`dm`] object's table.
+#' to reference the primary key column of another table of the [`dm`] object.
 #'
 #' @inheritParams cdm_add_fk
-#' @param table The table whose columns should be tested for suitability of use as foreign keys.
+#' @param table The table whose columns should be tested for suitability as foreign keys.
 #' @param ref_table A table with a primary key.
 #'
-#' @details `cdm_enum_fk_candidates()` first checks if `ref_table` has a primary key set.
-#' Then, for each column of `table`, a join operation will be tried, with argument `by` matching
-#' the respective column with the primary key of `ref_table`.
-#' This tests implicitly for
-#' type compatibility (on most sources).
-#' Based on the result of the join, the
-#' entry in the result column `why` is:
+#' @details `cdm_enum_fk_candidates()` first checks if `ref_table` has a primary key set,
+#' if not, an error is thrown.
 #'
-#' - an empty entry, if the column is a candidate
-#' - the total percentage and individual numbers of missing matches between the entries of the
-#' respective column in table `table` and the primary column entries in table `ref_table`.
-#' - the error message triggered by the error (often stating the mismatched column types)
+#' If `ref_table` does have a primary key, then a join operation will be tried using
+#' that key as the `by` argument of join() to match it to each column of `table`.
+#'
+#' On most data sources, this operation implicitly also tests for type compatibility
+#' between the columns of the two tables.
+#'
+#' The outcome of the join operation determines the value of the `why` column in the result:
+#'
+#' - an empty value for a column of `table` that is a suitable foreign key candidate
+#' - the count and percentage of missing matches for a column that is not suitable
+#' - a triggered error message that may include the types of mismatched columns
 #'
 #' @return A table that lists which columns of `table` would be suitable candidates for
```

Comment:
The table lists all columns. Maybe something like:

A table that lists, for all columns, whether they are suitable ... . For columns that are not suitable, the reason is explained in the `why` column.